### PR TITLE
Reduce the number of JSON-RPC calls in UI

### PR DESCRIPTION
**Pull request by k1rill-fedoseev**

### CODE REVIEW

File: ui/src/stores/HomeStore.js [hunk 1]
```diff
@@ -403,12 +406,14 @@ class HomeStore {
 
   @action
   async filterByTxHashInReturnValues(transactionHash) {
-    const events = await this.getEvents(1, 'latest')
+    const { blockNumber } = await this.homeWeb3.eth.getTransaction(transactionHash)
```
Comment (akolotov) [inline ui/src/stores/HomeStore.js:409]: What if the transaction is not mined yet?

**Review by akolotov — COMMENTED** [review]
Did you check this changes on the real deposit and withdrawal? Does the UI check the transaction status properly?

File: ui/src/stores/HomeStore.js (hunk 1 above)
Comment (k1rill-fedoseev) [reply ui/src/stores/HomeStore.js]: Indeed, there was a mistake. I have fixed it, and checked the ui in local testnet, everything seems be working fine.

File: ui/src/stores/HomeStore.js [hunk 2]
```diff
@@ -406,14 +406,17 @@ class HomeStore {
 
   @action
   async filterByTxHashInReturnValues(transactionHash) {
-    const { blockNumber } = await this.homeWeb3.eth.getTransaction(transactionHash)
-    const events = await this.getEvents(blockNumber, blockNumber)
+    const events = await this.getEvents(1, 'latest')
```
Comment (akolotov) [inline ui/src/stores/HomeStore.js]: Does it means that the startBlock is `1`? Do you think it makes sense to choose more relevant block? E.g. a block that was mined a day ago from the current moment or something like this?

Comment (k1rill-fedoseev) [reply ui/src/stores/HomeStore.js]: But what if a user want to access the events for the transaction happened more than 1 day ago?

Comment (akolotov) [reply ui/src/stores/HomeStore.js]: Is this for scenario when a user accesses a tx in the event tab? Will this tab display not mined transactions? If it is not from this tab, how the will the user access this transaction?

File: ui/src/stores/ForeignStore.js [hunk 3]
```diff
@@ -168,9 +169,9 @@ class ForeignStore {
     setInterval(() => {
       this.getBlockNumber()
       this.getEvents()
```
Comment (akolotov) [inline ui/src/stores/ForeignStore.js:171]: For which scenario are data/actions produced getEvents() used? Is it worth to reduce amount of requests to frame the getting events only for particular cases?

File: ui/src/stores/HomeStore.js (hunk 2 above)
Comment (k1rill-fedoseev) [reply ui/src/stores/HomeStore.js]: This function is used for searching home side events which include transactionHash on the foreign side in their data (such as `AffirmationCompleted`, `SignedForAffirmation`)

**Review by akolotov — APPROVED** [review]

**Pull request merged**